docs/okeps/okep-5521-single-cluster-host-dpu.md

```diff
@@ -0,0 +1,308 @@
+    k8s.ovn.org/node-subnets: '{"default":["10.244.2.0/24"]}'
+    k8s.ovn.org/node-transit-switch-port-ifaddr: '{"ipv4":"100.88.0.3/16"}'
+    k8s.ovn.org/remote-zone-migrated: ovn-worker
+    k8s.ovn.org/zone-name: ovn-worker
+    kubeadm.alpha.kubernetes.io/cri-socket: unix:///run/containerd/containerd.sock
+    node.alpha.kubernetes.io/ttl: "0"
+    volumes.kubernetes.io/controller-managed-attach-detach: "true"
+  creationTimestamp: "2025-08-25T22:12:56Z"
+  labels:
+    beta.kubernetes.io/arch: amd64
+    beta.kubernetes.io/os: linux
+    k8s.ovn.org/zone-name: ovn-worker
+    kubernetes.io/arch: amd64
+    kubernetes.io/hostname: ovn-worker
+    kubernetes.io/os: linux
+  name: ovn-worker
+  resourceVersion: "102658"
+  uid: 48ea7c29-ee34-48c1-b8a9-43e3db34c007
+spec:
+  podCIDR: 10.244.2.0/24
+  podCIDRs:
+  - 10.244.2.0/24
+```
+
+OVNK-Full DPU will create a new Node and update as needed:
+
+```yaml
+apiVersion: v1
+kind: Node
+metadata:
+  annotations:
+    k8s.ovn.org/identity: dpu1-ovnk
+    k8s.ovn.org/host-cidrs: '["172.18.0.3/16","172.19.0.2/16"]'
+    k8s.ovn.org/l3-gateway-config: '{"default":{"mode":"shared","bridge-id":"breth0","interface-id":"breth0_ovn-worker","mac-address":"c6:a8:ac:29:8c:d0","ip-addresses":["172.18.0.3/16"],"ip-address":"172.18.0.3/16","next-hops":["172.18.0.1"],"next-hop":"172.18.0.1","node-port-enable":"true","vlan-id":"0"}}'
+    k8s.ovn.org/node-chassis-id: 7f3c4b26-5d1a-4d7f-b3a9-9c52c1b6e0d2
+    k8s.ovn.org/node-encap-ips: '["10.0.0.4"]'
+    k8s.ovn.org/node-id: "3"
+    k8s.ovn.org/node-masquerade-subnet: '{"ipv4":"169.254.0.0/17","ipv6":"fd69::/112"}'
+    k8s.ovn.org/node-primary-ifaddr: '{"ipv4":"172.18.0.3/16"}'
+    k8s.ovn.org/node-subnets: '{"default":["10.244.2.0/24"]}'
+    k8s.ovn.org/node-transit-switch-port-ifaddr: '{"ipv4":"100.88.0.3/16"}'
+    k8s.ovn.org/remote-zone-migrated: ovn-worker-dpu1-ovnk
+    k8s.ovn.org/zone-name: ovn-worker
+    kubeadm.alpha.kubernetes.io/cri-socket: unix:///run/containerd/containerd.sock
+    node.alpha.kubernetes.io/ttl: "0"
+    volumes.kubernetes.io/controller-managed-attach-detach: "true"
+  creationTimestamp: "2025-08-25T22:12:56Z"
+  labels:
+    beta.kubernetes.io/arch: amd64
+    beta.kubernetes.io/os: linux
+    k8s.ovn.org/zone-name: ovn-worker
+    kubernetes.io/arch: amd64
+    kubernetes.io/hostname: ovn-worker
+    kubernetes.io/os: linux
+  name: ovn-worker-dpu1-ovnk
+  resourceVersion: "102658"
+  uid: 48ea7c29-ee34-48c1-b8a9-43e3db34c007
+spec:
+  podCIDR: 11.244.2.0/24
+  podCIDRs:
+  - 11.244.2.0/24
+```
+
+In the new CR, a new annotation is added "k8s.ovn.org/identity", this will be used by OVNK-Full DPU controllers
+to filter only on these node events. OVNK-Full Host will ignore nodes that have this identity field populated.
+
+TODO: figure out which values in the above would actually change.
+
+The node status will remain down, as kubelet will not be running in the DPU. But this status should be ignored by OVNK-Full DPU.
```

Comment:
I am concerned this can spook existing monitoring - or require its adjustment to special-handle these fake "Nodes". I wonder if instead of claiming the Node down we could make it resource-less (no memory or cpu capacity) - would it then make the scheduler smart enough to not attempt to put pods on these "Nodes"?

Reply:
yeah there are a few ways we can just mark the nodes so no pods can ever be scheduled